Added MiniIsolation for Electrons and Muons using the SUSY prescription

jkarancs opened the pull request:
Dear B2G,

In this commit I added the computation of the MiniIsolation for Electrons/Muons using the SUSY prescription posted here:
https://hypernews.cern.ch/HyperNews/CMS/get/susy/1991.html
code:
https://github.com/manuelfs/CfANtupler/blob/master/minicfa/interface/miniAdHocNTupler.h#L54-L120

Best regards,
Janos Karancsi

jkarancs committed:
Added MiniIsolation for Electrons and Muons using the SUSY prescription

decosa commented:
Hi Janos,

I have merged your changes, but if I checkout the branch and run the b2g cfg file I get this error:

----- Begin Fatal Exception 11-Aug-2015 20:31:38 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing run: 251162 lumi: 18 event: 13712798
   [1] Running path 'endPath'
   [2] Calling event method for module PoolOutputModule/'edmNtuplesOut'
   [3] Calling produce method for unscheduled module CandViewNtpProducer/'genJetsAK8'
   [4] Calling produce method for unscheduled module FastjetJetProducer/'ak8GenJetsNoNu'
   [5] Calling produce method for unscheduled module CandPtrSelector/'packedGenParticlesForJetsNoNu'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: reco::Candidate
Looking for module label: packedGenParticles
Looking for productInstanceName:


Please, can you have a look into it?

Thanks,

Annapaola


On 10 Aug 2015, at 15:37, János Karancsi <notifications@github.com<mailto:notifications@github.com>> wrote:


Dear B2G,

In this commit I added the computation of the MiniIsolation for Electrons/Muons using the SUSY prescription posted here:
https://hypernews.cern.ch/HyperNews/CMS/get/susy/1991.html
code:
https://github.com/manuelfs/CfANtupler/blob/master/minicfa/interface/miniAdHocNTupler.h#L54-L120

Best regards,
Janos Karancsi

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/cmsb2g/B2GAnaFW/pull/16

Commit Summary

  *   Added MiniIsolation for Electrons and Muons using the SUSY prescription

File Changes

  *   M python/b2gedmntuples_cff.py<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-0> (8)
  *   M src/ElectronUserData.cc<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-1> (12)
  *   A src/Isolations.cc<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-2> (74)
  *   A src/Isolations.h<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-3> (6)
  *   M src/MuonUserData.cc<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-4> (12)
  *   M test/b2gedmntuples_cfg.py<https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-5> (2)

Patch Links:

  *   https://github.com/cmsb2g/B2GAnaFW/pull/16.patch
  *   https://github.com/cmsb2g/B2GAnaFW/pull/16.diff

—
Reply to this email directly or view it on GitHub<https://github.com/cmsb2g/B2GAnaFW/pull/16>.

jkarancs commented:
Dear Annapaola,

I check it, but I think I already had this error when I ran on Data with
isData=False. :)
I succesffuly ran on Data and MC when i added the commit. Can you check if
that is the case?
In the mean time I try to reproduce this error again.

Best regards,
Janos

On 11 August 2015 at 20:39, Annapaola de Cosa <notifications@github.com>
wrote:

> Hi Janos,
>
> I have merged your changes, but if I checkout the branch and run the b2g
> cfg file I get this error:
>
> ----- Begin Fatal Exception 11-Aug-2015 20:31:38
> CEST-----------------------
> An exception of category 'ProductNotFound' occurred while
> [0] Processing run: 251162 lumi: 18 event: 13712798
> [1] Running path 'endPath'
> [2] Calling event method for module PoolOutputModule/'edmNtuplesOut'
> [3] Calling produce method for unscheduled module
> CandViewNtpProducer/'genJetsAK8'
> [4] Calling produce method for unscheduled module
> FastjetJetProducer/'ak8GenJetsNoNu'
> [5] Calling produce method for unscheduled module
> CandPtrSelector/'packedGenParticlesForJetsNoNu'
> Exception Message:
> Principal::getByToken: Found zero products matching all criteria
> Looking for a container with elements of type: reco::Candidate
> Looking for module label: packedGenParticles
> Looking for productInstanceName:
>
>
> Please, can you have a look into it?
>
> Thanks,
>
> Annapaola
>
>
> On 10 Aug 2015, at 15:37, János Karancsi <notifications@github.com<mailto:
> notifications@github.com>> wrote:
>
>
> Dear B2G,
>
> In this commit I added the computation of the MiniIsolation for
> Electrons/Muons using the SUSY prescription posted here:
> https://hypernews.cern.ch/HyperNews/CMS/get/susy/1991.html
> code:
>
> https://github.com/manuelfs/CfANtupler/blob/master/minicfa/interface/miniAdHocNTupler.h#L54-L120
>
> Best regards,
> Janos Karancsi
>
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/cmsb2g/B2GAnaFW/pull/16
>
> Commit Summary
>
> * Added MiniIsolation for Electrons and Muons using the SUSY prescription
>
> File Changes
>
> * M python/b2gedmntuples_cff.py<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-0> (8)
> * M src/ElectronUserData.cc<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-1> (12)
> * A src/Isolations.cc<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-2> (74)
> * A src/Isolations.h<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-3> (6)
> * M src/MuonUserData.cc<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-4> (12)
> * M test/b2gedmntuples_cfg.py<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-5> (2)
>
> Patch Links:
>
> * https://github.com/cmsb2g/B2GAnaFW/pull/16.patch
> * https://github.com/cmsb2g/B2GAnaFW/pull/16.diff
>
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/cmsb2g/B2GAnaFW/pull/16>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cmsb2g/B2GAnaFW/pull/16#issuecomment-130008670>.
>

decosa commented:
Hi Janos,

That was the case! The test file was supposed to be a MC sample.
I guess I committed a change to the input file by mistake while testing it on data.

Thanks,

Annapaola

On 11 Aug 2015, at 20:50, János Karancsi <notifications@github.com<mailto:notifications@github.com>> wrote:

Dear Annapaola,

I check it, but I think I already had this error when I ran on Data with
isData=False. :)
I succesffuly ran on Data and MC when i added the commit. Can you check if
that is the case?
In the mean time I try to reproduce this error again.

Best regards,
Janos

On 11 August 2015 at 20:39, Annapaola de Cosa <notifications@github.com<mailto:notifications@github.com>>
wrote:

> Hi Janos,
>
> I have merged your changes, but if I checkout the branch and run the b2g
> cfg file I get this error:
>
> ----- Begin Fatal Exception 11-Aug-2015 20:31:38
> CEST-----------------------
> An exception of category 'ProductNotFound' occurred while
> [0] Processing run: 251162 lumi: 18 event: 13712798
> [1] Running path 'endPath'
> [2] Calling event method for module PoolOutputModule/'edmNtuplesOut'
> [3] Calling produce method for unscheduled module
> CandViewNtpProducer/'genJetsAK8'
> [4] Calling produce method for unscheduled module
> FastjetJetProducer/'ak8GenJetsNoNu'
> [5] Calling produce method for unscheduled module
> CandPtrSelector/'packedGenParticlesForJetsNoNu'
> Exception Message:
> Principal::getByToken: Found zero products matching all criteria
> Looking for a container with elements of type: reco::Candidate
> Looking for module label: packedGenParticles
> Looking for productInstanceName:
>
>
> Please, can you have a look into it?
>
> Thanks,
>
> Annapaola
>
>
> On 10 Aug 2015, at 15:37, János Karancsi <notifications@github.com<mailto:notifications@github.com><mailto:
> notifications@github.com<mailto:notifications@github.com>>> wrote:
>
>
> Dear B2G,
>
> In this commit I added the computation of the MiniIsolation for
> Electrons/Muons using the SUSY prescription posted here:
> https://hypernews.cern.ch/HyperNews/CMS/get/susy/1991.html
> code:
>
> https://github.com/manuelfs/CfANtupler/blob/master/minicfa/interface/miniAdHocNTupler.h#L54-L120
>
> Best regards,
> Janos Karancsi
>
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/cmsb2g/B2GAnaFW/pull/16
>
> Commit Summary
>
> * Added MiniIsolation for Electrons and Muons using the SUSY prescription
>
> File Changes
>
> * M python/b2gedmntuples_cff.py<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-0> (8)
> * M src/ElectronUserData.cc<http://electronuserdata.cc/><
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-1> (12)
> * A src/Isolations.cc<http://isolations.cc/><
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-2> (74)
> * A src/Isolations.h<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-3> (6)
> * M src/MuonUserData.cc<http://muonuserdata.cc/><
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-4> (12)
> * M test/b2gedmntuples_cfg.py<
> https://github.com/cmsb2g/B2GAnaFW/pull/16/files#diff-5> (2)
>
> Patch Links:
>
> * https://github.com/cmsb2g/B2GAnaFW/pull/16.patch
> * https://github.com/cmsb2g/B2GAnaFW/pull/16.diff
>
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/cmsb2g/B2GAnaFW/pull/16>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cmsb2g/B2GAnaFW/pull/16#issuecomment-130008670>.
>

—
Reply to this email directly or view it on GitHub<https://github.com/cmsb2g/B2GAnaFW/pull/16#issuecomment-130015285>.
